source/grid/grid_in.cc

```diff
@@ -757,6 +757,31 @@ void GridIn<dim, spacedim>::read_ucd (std::istream &in)
   tria->create_triangulation_compatibility (vertices, cells, subcelldata);
 }
 
+namespace
+{
+  // Forward declaration of helper class
+  template <int dim>
+  class Abaqus_to_UCD;
+}
+
+template <int dim, int spacedim>
+void GridIn<dim, spacedim>::read_abaqus (std::istream &in)
+{
+  Assert (tria != 0, ExcNoTriangulationSelected());
+  Assert (dim==2 || dim==3, ExcNotImplemented());
+  AssertThrow (in, ExcIO());
+
+  // Read in the Abaqus file into an intermediate object
+  // that is to be passed along to the UCD reader
+  Abaqus_to_UCD<dim> abaqus_to_ucd;
+  const bool success_abaqus_in = abaqus_to_ucd.read_in_abaqus(in);
+  AssertThrow(success_abaqus_in == true, ExcInternalError());
```

Comment:
Thanks -- within the limits of your time. If the code was bad before, I'm not asking you to rewrite it.